contracts/modules/common/BaseFeature.sol

```diff
@@ -0,0 +1,178 @@
+import "../../infrastructure/IModuleRegistry.sol";
+import "../../infrastructure/storage/ILockStorage.sol";
+import "./IFeature.sol";
+import "../../../lib/other/ERC20.sol";
+import "./IVersionManager.sol";
+
+/**
+ * @title BaseFeature
+ * @notice Base Feature contract that contains methods common to all Feature contracts.
+ * @author Julien Niset - <julien@argent.xyz>, Olivier VDB - <olivier@argent.xyz>
+ */
+contract BaseFeature is IFeature {
+
+    // Empty calldata
+    bytes constant internal EMPTY_BYTES = "";
+    // Mock token address for ETH
+    address constant internal ETH_TOKEN = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    // The address of the module registry.
+    IModuleRegistry internal registry;
+    // The address of the Lock storage
+    ILockStorage internal lockStorage;
+    // The address of the Version Manager
+    IVersionManager internal versionManager;
+
+    event FeatureCreated(bytes32 name);
+    event FeatureInitialised(address wallet);
+
+    /**
+     * @notice Throws if the wallet is locked.
+     */
+    modifier onlyWhenUnlocked(address _wallet) {
+        require(!lockStorage.isLocked(_wallet), "BF: wallet locked");
+        _;
+    }
+
+    /**
+     * @notice Throws if the sender is not the target wallet of the call.
+     */
+    modifier onlyWallet(address _wallet) {
+        require(msg.sender == _wallet, "BF: caller must be wallet");
+        _;
+    }
+
+    /**
+     * @notice Throws if the sender is not the VersionManager.
+     */
+    modifier onlyVersionManager() {
+        require(msg.sender == address(versionManager), "BF: caller must be VersionManager");
+        _;
+    }
+
+    /**
+     * @notice Throws if the sender is not the owner of the target wallet.
+     */
+    modifier onlyWalletOwner(address _wallet) {
+        require(isOwner(_wallet, msg.sender), "BF: must be wallet owner");
+        _;
+    }
+
+    /**
+     * @notice Throws if the sender is not an authorised feature of the target wallet.
+     */
+    modifier onlyWalletFeature(address _wallet) {
+        require(versionManager.isFeatureAuthorised(_wallet, msg.sender), "BF: must be a wallet feature");
+        _;
+    }
+
+    /**
+     * @notice Throws if the sender is not the owner of the target wallet or the feature itself.
+     */
+    modifier onlyWalletOwnerOrFeature(address _wallet) {
+        // Wrapping in an internal method reduces deployment cost by avoiding duplication of inlined code
+        verifyOwnerOrAuthorisedFeature(_wallet, msg.sender);
+        _;
+    }
+
+    constructor(
+        IModuleRegistry _registry,
+        ILockStorage _lockStorage,
+        IVersionManager _versionManager,
+        bytes32 _name
+    ) public {
+        registry = _registry;
+        lockStorage = _lockStorage;
+        versionManager = _versionManager;
+        emit FeatureCreated(_name);
+    }
+
+    /**
+    * @inheritdoc IFeature
+    */
+    function recoverToken(address _token) external override {
```

Comment:
Tokens should be transferred to the `VersionManager` which would remove the dependency of the Features on the `ModuleRegistry`. We can then remove it from the constructor of all the features.

Reply:
We will need to add 
```
    function recoverToken(address _token) external onlyOwner {
        uint total = ERC20(_token).balanceOf(address(this));
        ERC20(_token).transfer(msg.sender, total);
    }
```
to the `VersionManager`.

Reply:
Done